lib/src/main/java/ua/naiksoftware/stomp/StompClient.java

```diff
@@ -240,11 +240,12 @@ public Completable disconnectCompletable() {
             return Flowable.error(new IllegalArgumentException("Topic path cannot be null"));
         else if (!streamMap.containsKey(destPath))
             streamMap.put(destPath,
-                    subscribePath(destPath, headerList).andThen(
+                    Completable.defer(() -> subscribePath(destPath, headerList)).andThen(
                     getMessageStream()
                             .filter(msg -> pathMatcher.matches(destPath, msg))
                             .toFlowable(BackpressureStrategy.BUFFER)
-                            .share()).doFinally(() -> unsubscribePath(destPath).subscribe())
+                            .doFinally(() -> unsubscribePath(destPath).subscribe())
```

Comment:
Unsubscribe before the share to allow this to only occur once all Disposables have been disposed for this topic. If you unsubscribe after the share every single Disposable will cause `unsubscribe(destPath)` to be called.